libs/books/data-access/src/lib/+state/reading-list.reducer.ts

```diff
@@ -58,6 +58,9 @@ const readingListReducer = createReducer(
   ),
   on(ReadingListActions.failedRemoveFromReadingList, (state, action) =>
     readingListAdapter.addOne(action.item, state)
+  ),
+  on(ReadingListActions.confirmedMarkAsFinished, (state, action) =>
```

Comment:
Is this in keeping with the rest of the data handling patterns? We're using optimistic updates elsewhere, shouldn't we use the same here?

Reply:
I made this design choice because I want the backend to be the source of record for the `finishedDate` timestamp reflected in the UI. By making this action pessimistic, I can ensure the `finishedDate` displayed is the exact timestamp stored in the backend. I've worked with timestamps before and seen cases where it's important to have the exact value in the UI.

Reply:
This leaves us with an inconsistent UI though. How would you change this to make it optimistic?

Reply:
I updated the logic to make it optimistic